src/H5FDsubfiling/H5subfiling_common.c

```diff
@@ -310,13 +313,21 @@ H5_free_subfiling_object(int64_t object_id)
             H5_SUBFILING_GOTO_ERROR(H5E_VFL, H5E_CANTGET, FAIL,
                                     "couldn't get subfiling context for subfiling object ID");
 
-        if (H5_free_subfiling_object_int(sf_context) < 0)
-            H5_SUBFILING_GOTO_ERROR(H5E_VFL, H5E_CANTFREE, FAIL, "couldn't free subfiling context object");
+        if (sf_context->file_ref == 0 || --sf_context->file_ref == 0) {
```

Comment:
Check for a ref. of 0 here since we can end up failing during context object creation before it's been associated with a file.